server/jetstream_cluster.go

```diff
@@ -160,11 +162,37 @@ type consumerAssignment struct {
 	Reply   string          `json:"reply,omitempty"`
 	State   *ConsumerState  `json:"state,omitempty"`
 	// Internal
-	responded  bool
-	recovering bool
-	pending    bool
-	deleted    bool
-	err        error
+	responded       bool
+	recovering      bool
+	pending         bool
+	deleted         bool
+	err             error
+	unsupportedJson []byte // The raw JSON content of the assignment, if it's unsupported due to the required API level.
+}
+
+type writeableConsumerAssignment struct {
+	consumerAssignment
+	// Internal
+	unsupportedJson []byte // The raw JSON content of the assignment, if it's unsupported due to the required API level.
+}
+
+func (wca *writeableConsumerAssignment) MarshalJSON() ([]byte, error) {
+	if wca.unsupportedJson != nil {
+		return wca.unsupportedJson, nil
+	}
+	return json.Marshal(wca.consumerAssignment)
+}
+
+func (wca *writeableConsumerAssignment) UnmarshalJSON(data []byte) error {
+	var ca consumerAssignment
+	if err := json.Unmarshal(data, &ca); err != nil {
```

Comment:
We could go a step further here and decode the assignments in strict mode to catch unknown configuration fields too, WDYT?
```
dec := json.NewDecoder(bytes.NewReader(data))
dec.DisallowUnknownFields()
if err := dec.Decode(); err != nil { ... }
```

Reply:
Done.
We decode under strict and mark it as unsupported if it fails. We then still decode with `json.Unmarshal` to include as much info that's recognized to protect stream subjects space, respond to API requests, etc.